src/Orleans.Transactions/State/StorageBatch.cs

```diff
@@ -74,38 +74,43 @@ internal class StorageBatch<TState> : ITransactionalStateStorageEvents<TState>
         private int confirm = 0;
         private int collect = 0;
         private int cancel = 0;
-
+        private JsonSerializerSettings serializerSettings;
         public MetaData MetaData { get; private set; }
 
         public string ETag { get; set; }
 
         public int BatchSize => total;
-
         public override string ToString()
         {
             return $"batchsize={total} [{read}r {prepare}p {commit}c {confirm}cf {collect}cl {cancel}cc]";
         }
 
-        public StorageBatch(TransactionalStorageLoadResponse<TState> loadresponse)
+        public StorageBatch(TransactionalStorageLoadResponse<TState> loadresponse, JsonSerializerSettings settings)
         {
-            MetaData = ReadMetaData(loadresponse);
+            if(settings == null)
+                throw new ArgumentNullException(nameof(settings));
+            this.serializerSettings = settings;
+            MetaData = ReadMetaData(loadresponse, this.serializerSettings);
             ETag = loadresponse.ETag;
             confirmUpTo = loadresponse.CommittedSequenceId;
             cancelAbove = loadresponse.PendingStates.LastOrDefault()?.SequenceId ?? loadresponse.CommittedSequenceId;
             cancelAboveStart = cancelAbove;
         }
 
-        public StorageBatch(StorageBatch<TState> previous)
+        public StorageBatch(StorageBatch<TState> previous, JsonSerializerSettings settings)
         {
+            if (settings == null)
+                throw new ArgumentNullException(nameof(settings));
+            this.serializerSettings = settings;
```

Comment:
nit: - style comment, feel free to ignore
this.serializerSettings = settings ?? throw new ArgumentNullException(nameof(settings)); 
is less code and looks cleaner.